src/PlaywrightRunner.ts

```diff
@@ -81,67 +103,95 @@ class PlaywrightRunner extends JestRunner {
     const pwTests: Test[] = []
     for (const test of tests) {
-      for (const browser of browsers) {
-        checkBrowserEnv(browser)
-        const { devices: availableDevices, instance } = getPlaywrightInstance(
-          browser,
-        )
-        let wsEndpoint: WsEndpointType = connectOptions?.wsEndpoint || null
-        if (launchType === SERVER && wsEndpoint === null) {
-          if (!this.browser2Server[browser]) {
-            const options = getBrowserOptions(browser, launchOptions)
-            this.browser2Server[browser] = await instance.launchServer(options)
-          }
-          wsEndpoint = this.browser2Server[browser]!.wsEndpoint()
-        }
+      if (useDefaultBrowserType) {
+        const { devices: availableDevices, instance } = getPlaywrightInstance()
+        const resultDevices = getDevices(devices, availableDevices)
 
-        if (devices instanceof RegExp) {
-          resultDevices = Object.keys(availableDevices).filter((item) =>
-            item.match(devices),
-          )
-        } else {
-          if (devices) {
-            resultDevices = devices
-          }
+        const browserTest = {
+          test: test as JestPlaywrightTest,
+          config,
         }
-
         if (resultDevices.length) {
-          resultDevices.forEach((device: DeviceType) => {
+          for (const device of resultDevices) {
+            const browser =
+              typeof device === 'string'
+                ? availableDevices[device].defaultBrowserType
+                : CHROMIUM
+            let wsEndpoint: WsEndpointType = connectOptions?.wsEndpoint || null
+            await this.launchServer(
+              config,
+              wsEndpoint,
+              browser,
+              (instance as Record<BrowserType, GenericBrowser>)[browser],
+            )
+
+            wsEndpoint = this.browser2Server[browser]!.wsEndpoint()
+
             if (typeof device === 'string') {
               const availableDeviceNames = Object.keys(availableDevices)
               checkDeviceEnv(device, availableDeviceNames)
             }
+
             pwTests.push(
-              getBrowserTest(
-                test as JestPlaywrightTest,
-                config,
-                browser,
-                wsEndpoint,
-                device,
-              ),
+              getBrowserTest({ ...browserTest, browser, wsEndpoint, device }),
             )
-          })
-        } else {
-          pwTests.push(
-            getBrowserTest(
-              test as JestPlaywrightTest,
-              config,
-              browser,
-              wsEndpoint,
-              null,
-            ),
+          }
+        }
+      } else {
+        for (const browser of browsers) {
+          checkBrowserEnv(browser)
+          const { devices: availableDevices, instance } = getPlaywrightInstance(
+            browser,
+          )
+          const resultDevices = getDevices(devices, availableDevices)
```

Comment:
It seems, that there will be error if custom devices are used.

With such config
```
  browsers: ['chromium', 'firefox', 'webkit'],
  devices: [
    'iPad Pro 11 landscape',
    'iPhone SE',
    {
      name: 'FHD desktop',
      viewport: {
        width: 1920,
        height: 1080,
      },
      deviceScaleFactor: 1,
      isMobile: false,
      hasTouch: false,
      defaultBrowserType: 'webkit',
    },
    {
      name: 'FHD desktop',
      viewport: {
        width: 1920,
        height: 1080,
      },
      deviceScaleFactor: 1,
      isMobile: false,
      hasTouch: false,
      defaultBrowserType: 'firefox',
    },
  ],
```

Test will be runned on webkit-ipad, webkit-iphone, and two times on chromium desktop.

Reply:
#464 464